indico/modules/attachments/forms.py

```diff
@@ -80,8 +83,24 @@ class EditAttachmentLinkForm(AttachmentLinkFormMixin, EditAttachmentFormBase):
     pass
 
 
+def _prepare_material_types_list():
+    return [{'id': value, 'name': value} for value in material_types]
+
+
+class _FolderNameAutoCompleteField(HiddenField):
+    widget = StaticSelectizeWidget(choices=_prepare_material_types_list())
+
+    def process_formdata(self, valuelist):
+        if valuelist and valuelist[0]:
+            self.data = valuelist[0]
+
+    def process_data(self, value):
+        type_ = namedtuple('MaterialType', ['id', 'name'])
```

Comment:
namedtuples should be defined at import time; they are basically implicit class definitions